### ISSUE TITLE
502 Bad gateway while deploying to aws.

### ISSUE THREAD

**Issue by husain-nahar** [serverpod/serverpod#2858]
Serverpod version: 2.1.4
dart version: 3.5.3

I followed Changes made by @hidayet-poscu. But I am still getting the bad gateway error.
Why can't it be fixed once for all @vlidholt.
My whole server is stuck now and I cnt go further on my work.
#2620 #2650 didn't fix this up.
I want to know what I am missing here.
Do I need to re run terraform scripts?
my linux machine on aws is also having dart 3.5.3

**Comment by TanguyDeBrabandere** [comment]
I had a server running on AWS that had no issues whatsoever until this morning, without me doing any update at all. I've been trying to figure it out all morning, but I have absolutely no clue what can be wrong. Serverpod api seems to be up and running, but I also get 502 bad gateway.

**Comment by husain-nahar** [comment]
@TanguyDeBrabandere until serverpod 2.0 i had uploaded and pushed to aws for deploy many times but never faced such issue. But since i have updated to 2.0 and later i am starting to face 502 bad gateway after every deploy.

**Comment by Isakdl** [comment]
Hey @husain-nahar @TanguyDeBrabandere check out this thread: https://github.com/serverpod/serverpod/discussions/2829

In essence the most likely problem is the dart version you have on the server. Make sure to install a new dart version and delete the old one.

**Comment by husain-nahar** [comment]
@Isakdl I go with the latest dart version on server always whenever I change something from dart version 3.1.0 and Serverpod version 2.0.0, it fails. and shows bad gateway.

**Comment by adiltirur** [comment]
@husain-nahar can you show me what your run_serverpod file looks like? May be I can help you out.

**Comment by husain-nahar** [comment]
below is the run_serverpod file content:

#!/bin/sh
RUNMODE=$(cat /home/ec2-user/runmode)
SERVER_ID=$(curl -s http://[my private ip])
cd /home/ec2-user/serverpod/active/my_server
/usr/lib/dart/bin/dart --old_gen_heap_size=0  run bin/main.dill --mode $RUNMODE --server-id $SERVER_ID > /home/ec2-user/serverpod.log 2> /home/ec2-user/serverpod.err

**Comment by Isakdl** [comment]
@husain-nahar Did you make sure to delete the old dart version from the server? If not you may still be executing with the old runtime.

**Comment by husain-nahar** [comment]
yeah I check the dart version and it shows what I have installed latest on my server side.

**Comment by Isakdl** [comment]
Yes but did you also remove the old version, just installing the new doesn't automatically remove the old runtime.

**Comment by adiltirur** [comment]
@husain-nahar 

it’s likely that your server is using an outdated Dart version located at /usr/lib/dart/bin/dart.

To fix this, try the following updates:

In your run_serverpod file, replace the last line:

`/usr/lib/dart/bin/dart`

with:

`dart`

In your start_server file, replace:


`sudo -u ec2-user /usr/lib/dart/bin/dart pub get`

with:

`sudo -u ec2-user dart pub get`

This resolved a similar issue for me. However, to confirm, log into your server and check the Dart version with:


`dart --version`

Also, you can compare by running:


`/usr/lib/dart/bin/dart --version`

This will help you spot any differences.

**Comment by husain-nahar** [comment]
@Isakdl @adiltirur 
I followed what you said above @adiltirur and also followed below updating process of dart sdk:
echo "Installing dart"
wget -q https://storage.googleapis.com/dart-archive/channels/stable/release/3.5.4/sdk/dartsdk-linux-x64-release.zip
unzip -q dartsdk-linux-x64-release.zip 
sudo mv dart-sdk/ /usr/lib/dart/
NOTE: mv means to rename or remove. If there is error says dart-sdk already exist,
 Use 1st -
sudo rm -rf /usr/lib/dart/dart-sdk 
then repeat above step 4 again.
 it should now successfully move dart-sdk to usr/lib/dart.
  5. echo 'export PATH="/usr/lib/dart/dart-sdk/bin:$PATH"' | sudo tee -a /etc/profile.d/script.sh if above code is successfully done then you should see something like this: 
  export PATH="/usr/lib/dart/dart-sdk/bin:$PATH" 
  
  Now when I do dart --version it shows 3.5.4
  But when I do /usr/lib/dart/bin/dart --version it says 3.1.0
  
How to fix this please help.
@adiltirur what you said above I made those changes still I see 3.1.0

**Comment by husain-nahar** [comment]
when I check /usr/lib/dart/version it shows 3.1.0
but if I check /usr/lib/dart/dart-sdk/version it shows 3.5.4

I am confused on how to fix this.

**Comment by adiltirur** [comment]
@husain-nahar any easy fix would be search for 

 /usr/lib/dart/bin/dart
 
 in your codebase, this will show if anywhere you are still referring to old dart version, and instead of that just use
 
 dart
 
 This will help you solve the issue, and then rebuild and reploy.
 
 This would have caused because of of not updating your install_dependencie script.
 
 Please make sure you had updated the script.
 
 Follow https://docs.serverpod.dev/deployments/deploying-to-aws, in this look at Configure Dart Version.

**Comment by husain-nahar** [comment]
Following this link [https://docs.serverpod.dev/deployments/deploying-to-aws](url).
I was able to fix this issue as my serverpod version is now 2.1.5(latest) with dart version 3.5.4(latest).
But @Isakdl I am still getting 3.1.0 version when I cal /usr/lib/dart/version.
and if I check /usr/lib/dart/dart-sdk/version it shows 3.5.4
However the 502 bad gateway is gone for now.
Thanks @Isakdl @adiltirur for your time.
Closing this issue as  completed for now.
